static/js/apps/base/components/Footer.tsx

```diff
@@ -0,0 +1,123 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, useMemo } from "react";
+
+import { FooterMenu } from "../../../shared/types/base";
+import { Labels, Routes } from "../../../shared/types/general";
+import { resolveHref } from "../utilities/utilities";
+
+interface FooterProps {
+  hideFullFooter: boolean;
+  hideSubFooter: boolean;
+  subFooterExtra: string;
+  brandLogoLight: boolean;
+  footerMenu: FooterMenu[];
+  labels: Labels;
+  routes: Routes;
+}
+
+const Footer = ({
+  hideFullFooter,
+  hideSubFooter,
+  subFooterExtra,
+  brandLogoLight,
+  footerMenu,
+  labels,
+  routes,
+}: FooterProps): ReactElement => {
+  const visibleFooterMenu = useMemo(() => {
+    return footerMenu.map((footerMenuItem) => ({
+      ...footerMenuItem,
+      subMenu: footerMenuItem.subMenu.filter(
+        (filterSubMenuItem) => !filterSubMenuItem.hide
+      ),
+    }));
+  }, [footerMenu]);
+
+  return (
+    <footer id="main-footer">
+      {!hideFullFooter && visibleFooterMenu.length > 0 && (
+        <div className="container top-footer">
+          <div className="row">
+            {visibleFooterMenu.map((footerMenuItem) => (
+              <section
+                key={footerMenuItem.label}
+                className="col-12 col-sm-6 col-md-4"
+              >
+                <h6>{labels[footerMenuItem.label]}</h6>
+                {footerMenuItem.subMenu.map((footerSubMenuItem) => (
+                  <a
+                    key={footerSubMenuItem.label}
+                    href={resolveHref(footerSubMenuItem.href, routes)}
+                  >
+                    <span className="material-icons-outlined">
+                      arrow_forward
+                    </span>
+                    <span>{labels[footerSubMenuItem.label]}</span>
+                  </a>
+                ))}
+              </section>
+            ))}
+          </div>
+        </div>
+      )}
+      {!hideSubFooter && (
+        <div id="sub-footer">
+          <div className="container">
+            <span className="brand-byline">
+              <span className="brand-text">{labels["An initiative from"]}</span>
+              <img
+                className="brand-logo"
+                width="74"
+                height="25"
+                src={
+                  brandLogoLight
+                    ? "/images/google-logo-reverse.svg"
+                    : "/images/google-logo.svg"
+                }
+                alt="Google logo"
+              />
+              <img
+                className="brand-logo"
+                width="74"
+                height="25"
+                src="/images/google-logo-reverse.svg"
+                alt="Google logo"
+              />
+            </span>
+            <div className="sub-footer-links">
+              {subFooterExtra && (
+                <div dangerouslySetInnerHTML={{ __html: subFooterExtra }} />
```

Comment:
This looks like it gets set with a `getElementById` in `main.ts` and is "safe" HTML, but I don't see any usage of it anywhere. Maybe in some of the forks? Now that we are using React everywhere we should try to minimize injection of HTML if possible. In any case, this looks perfectly converted so nothing to do about it now, but I've added it to the list of possible future improvements.